### PR TITLE
Add an entry in the readme files explaining how to add an HSTS header

### LINKED ISSUE #18: Add support for HTTP Strict Transport Security (HSTS)
As a site admin, I should have a simple way to enable HSTS on my site.

- [ ] AC1: Assuming HTTPS checks all clearly show I probably can enable the HSTS header, I should be presented with this option in the HTTPS dialogue in #17.
- [ ] AC2: If HSTS activation fails during a loopback request, warn the user and prevent them from activating this setting.

---

A best practice for HTTPS is to also use [HTTP Strict Transport Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) (HSTS):

> If a website accepts a connection through HTTP and redirects to HTTPS, visitors may initially communicate with the non-encrypted version of the site before being redirected, if, for example, the visitor types http://www.foo.com/ or even just foo.com. This creates an opportunity for a man-in-the-middle attack. The redirect could be exploited to direct visitors to a malicious site instead of the secure version of the original site.
> 
> The HTTP Strict Transport Security header informs the browser that it should never load a site using HTTP and should automatically convert all attempts to access the site using HTTP to HTTPS requests instead.

If we detect HTTPS support (#6) and the user opts-in to use it (#17) then in addition to redirecting from HTTP to HTTPS then we should also serve the HSTS header.

See also core ticket [#28520](https://core.trac.wordpress.org/ticket/28520).

### PR DESCRIPTION
**Pull request by kienstra**
**Request For Review**

Hi @westonruter,
Following up on our conversation in https://github.com/xwp/pwa-wp/pull/82#issuecomment-430828370, this adds an entry in the `readme` files about how to add an `HSTS` header.

I'm not sure if this what you had in mind for documentation, though.

Thanks, Weston!

Closes #18.

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -157,6 +157,25 @@ HTTPS is a prerequisite for progressive web apps. A service worker is only able
...
+You can optionally add an [HSTS header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) (HTTP `Strict-Transport-Security`). This indicates to the browser to only load the site with HTTPS, not HTTP.
+
+<pre lang=php>
+/**
+ * Adds an HSTS header to the response.
+ *
+ * @param array $headers The headers to filter.
+ */
+add_filter( 'wp_headers', function( $headers ) {
+	$headers['Strict-Transport-Security'] = 'max-age=3600'; // Or another max-age.
```
Comment (westonruter) [inline readme.txt]: This needs to return the `$headers` array.

File: readme.txt [hunk 2]
```diff
@@ -157,6 +157,25 @@ HTTPS is a prerequisite for progressive web apps. A service worker is only able
 
 At the moment the plugin provides an API to detection of whether a site supports HTTPS. Building on that it's intended that this can then be used to present a user with an opt-in to switch over to HTTPS, which will also then need to include support for rewriting URLs from HTTP to HTTPS. See [labeled GitHub issues](https://github.com/xwp/pwa-wp/issues?q=label%3Ahttps) and see WordPress core tracking ticket [#28521](https://core.trac.wordpress.org/ticket/28521).
 
+You can optionally add an [HSTS header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) (HTTP `Strict-Transport-Security`). This indicates to the browser to only load the site with HTTPS, not HTTP.
+
+<pre lang=php>
+/**
+ * Adds an HSTS header to the response.
+ *
+ * @param array $headers The headers to filter.
```
Comment (westonruter) [inline readme.txt]: Add a `@return` tag.

File: readme.txt (hunk 1 above)
Comment (kienstra) [reply readme.txt]: That's a big omission 😄 
https://github.com/xwp/pwa-wp/pull/100/commits/dd49a54cf4009a2b77a74809e4a3c349d57223ca adds the `return` statemetn.

File: readme.txt (hunk 2 above)
Comment (kienstra) [reply readme.txt]: Thanks, https://github.com/xwp/pwa-wp/pull/100/commits/dd49a54cf4009a2b77a74809e4a3c349d57223ca adds the `@return` tag.

**Pull request merged**